Scrum 27 and 3

ethantankw opened the pull request:
Pull request for scrum 27 and 3

ethantankw committed:
Scrum 27 and 3

View arrangement requests, filter arrangement requests

ethantankw committed:
Large amount of submitted requests

Updated to include large batch of submitted requests

ryanang333 commented on backend/api/controllers/submitted_viewController.js:
actually, do we need another controller and route for this certain functionality?

since this is a get request for the "ArrangementRequests" model, can we add the route to the "ArrangementRequest" path and reuse the same controller?

ryanang333 replied:
based on SCRUM-23, it seems like this story should be to "view all requests that the staff has submitted", 

could you clarify whether the query you have
"  const arrangementRequests = await ArrangementRequest.find()
    .populate('staff');"
    
   is achieving this functionality? - it seems like its just fetching ALL arrangementrequests, but shouldnt it be filtering by the arrangementrequests where the "staff_id" matches the current staff_id?
   
   i.e., 
   await MyModel.find({ name: 'john', age: { $gte: 18 } }).exec();

   const arrangementRequests = await ArrangementRequest.find({staff_id: staffId})

ryanang333 replied:
![image](https://github.com/user-attachments/assets/bdee238a-1cf7-47ad-933e-e6420848dae6)
also, i think there are some conflicts. could you resolve them in your branch and raise another mr?

ethantankw committed:
Merge branch 'main' into Scrum-27&3

ethantankw committed:
Check based on staff_id

Added check based on staff_id (hardcoded for now as log in user story not done yet)
